src/core/qgsfieldformatter.h

```diff
@@ -112,7 +119,26 @@ class CORE_EXPORT QgsFieldFormatter
      */
     virtual QList< QgsVectorLayerRef > layerDependencies( const QVariantMap &config ) const SIP_SKIP;
 
+    /**
+     * Returns a list of the values that would be possible to select with this widget type
+     * On a RelationReference that would be the parents ids or on ValueMap all the configured keys
+     * according to the settings in the \a config
+     * \since QGIS 3.12
+     */
+    virtual QList<QVariant> availableValues( const QVariantMap &config, int countLimit ) const;
```

Comment:
I would suggest adding a new FieldFormatterContext class and argument here, with the FieldFormatterContext containining a getter/setter for the project for now (or just the relation manager?). It would avoid the introduction of the QgsProject::instance() use, and make extending this in future to be much easier without breaking or complicating the API.